pageserver/src/tenant/timeline.rs

```diff
@@ -3175,9 +3204,10 @@ impl Timeline {
             anyhow::bail!("cannot flush frozen layers when flush_loop is not running, state is {flush_loop_state:?}")
         }
 
-        self.layer_flush_start_tx.send_modify(|counter| {
+        self.layer_flush_start_tx.send_modify(|(counter, lsn)| {
             my_flush_request = *counter + 1;
             *counter = my_flush_request;
+            *lsn = to_lsn;
         });
```

Comment:
The `tokio::sync::watch` make ssense for the counter semantics, which is strongly monotonic.

I don't understand what the semantics for `lsn` are. Does it have to be monotonic? What if it's not?  Should we assert monotonicity here?
=> doc comment of the `layer_flush_start_tx` field should cover it.